### ISSUE TITLE
Alternate logging format

### ISSUE THREAD

**Issue by igor-dmitriev** [p6spy/p6spy#302]
Here is my code :
    @Transactional
    public void test() {
        List<Client> list = em.createQuery("select c from Client c inner join c.accounts", Client.class).getResultList();
        for (Client client : list) {
            client.setName("Igor");
        }
    }

database state :
<img width="437" alt="screen shot 2015-11-07 at 12 04 10" src="https://cloud.githubusercontent.com/assets/4592740/11014604/d8fc2cc8-8546-11e5-82ff-96ebd901ad74.png">
<img width="225" alt="screen shot 2015-11-07 at 12 04 26" src="https://cloud.githubusercontent.com/assets/4592740/11014607/f498aef2-8546-11e5-89a8-9067c5143366.png">

When I ran my code I got logs :
12:06:12|40|statement|connection 1|select relname from pg_class where relkind='S'|
12:06:13|28|statement|connection 2|/\* select c from Client c inner join c.accounts _/ select client0_.id_client as id_clien1_2_, client0_.age as age2_2_, client0_.name as name3_2_, client0_.nick as nick4_2_ from CLIENT client0_ inner join ACCOUNT accounts1_ on client0_.id_client=accounts1_.client_id|/_ select c from Client c inner join c.accounts _/ select client0_.id_client as id_clien1_2_, client0_.age as age2_2_, client0_.name as name3_2_, client0_.nick as nick4_2_ from CLIENT client0_ inner join ACCOUNT accounts1_ on client0_.id_client=accounts1_.client_id
12:06:13|1|statement|connection 2|select accounts0_.client_id as client_i3_2_0_, accounts0_.account_id as account_1_0_0_, accounts0_.account_id as account_1_0_1_, accounts0_.AMOUNT as AMOUNT2_0_1_, accounts0_.client_id as client_i3_0_1_ from ACCOUNT accounts0_ where accounts0_.client_id=?|select accounts0_.client_id as client_i3_2_0_, accounts0_.account_id as account_1_0_0_, accounts0_.account_id as account_1_0_1_, accounts0_.AMOUNT as AMOUNT2_0_1_, accounts0_.client_id as client_i3_0_1_ from ACCOUNT accounts0_ where accounts0_.client_id=5001
12:06:13|1|statement|connection 2|select accounts0_.client_id as client_i3_2_0_, accounts0_.account_id as account_1_0_0_, accounts0_.account_id as account_1_0_1_, accounts0_.AMOUNT as AMOUNT2_0_1_, accounts0_.client_id as client_i3_0_1_ from ACCOUNT accounts0_ where accounts0_.client_id=?|select accounts0_.client_id as client_i3_2_0_, accounts0_.account_id as account_1_0_0_, accounts0_.account_id as account_1_0_1_, accounts0_.AMOUNT as AMOUNT2_0_1_, accounts0_.client_id as client_i3_0_1_ from ACCOUNT accounts0_ where accounts0_.client_id=5000
12:06:13|5|statement|connection 2|/_ update com.luxoft.igor.entity.Client _/ update CLIENT set age=?, name=?, nick=? where id_client=?|/_ update com.luxoft.igor.entity.Client */ update CLIENT set age=79380, name='Igor', nick=NULL where id_client=5001
12:06:13|13|commit|connection 2||

Here is no second update statement, just for Client with id 5001, but lets browse Client table after test method was executed :
<img width="431" alt="screen shot 2015-11-07 at 12 07 49" src="https://cloud.githubusercontent.com/assets/4592740/11014612/53401f30-8547-11e5-9149-2d7603f5a51d.png">
Two clients were updated successfully !

And what happens when I turn off p6spy :
Hibernate: 
    /\* select
        c 
    from
        Client c 
    inner join
        c.accounts _/ select
            client0_.id_client as id_clien1_2_,
            client0_.age as age2_2_,
            client0_.name as name3_2_,
            client0_.nick as nick4_2_ 
        from
            CLIENT client0_ 
        inner join
            ACCOUNT accounts1_ 
                on client0_.id_client=accounts1_.client_id
Hibernate: 
    select
        accounts0_.client_id as client_i3_2_0_,
        accounts0_.account_id as account_1_0_0_,
        accounts0_.account_id as account_1_0_1_,
        accounts0_.AMOUNT as AMOUNT2_0_1_,
        accounts0_.client_id as client_i3_0_1_ 
    from
        ACCOUNT accounts0_ 
    where
        accounts0_.client_id=?
Hibernate: 
    select
        accounts0_.client_id as client_i3_2_0_,
        accounts0_.account_id as account_1_0_0_,
        accounts0_.account_id as account_1_0_1_,
        accounts0_.AMOUNT as AMOUNT2_0_1_,
        accounts0_.client_id as client_i3_0_1_ 
    from
        ACCOUNT accounts0_ 
    where
        accounts0_.client_id=?
Hibernate: 
    /_ update
        com.luxoft.igor.entity.Client _/ update
            CLIENT 
        set
            age=?,
            name=?,
            nick=? 
        where
            id_client=?
Hibernate: 
    /_ update
        com.luxoft.igor.entity.Client */ update
            CLIENT 
        set
            age=?,
            name=?,
            nick=? 
        where
            id_client=?

And as you see here are two update statements !! 

Here is my spy.properties :
driverlist=org.postgresql.Driver
dateformat=HH:mm:ss
appender=com.p6spy.engine.spy.appender.StdoutLogger
logMessageFormat=com.p6spy.engine.spy.appender.SingleLineFormat
excludecategories=info,debug,result,resultset,batch

jdbc properties :
jdbc.driverClassName=com.p6spy.engine.spy.P6SpyDriver
jdbc.url=jdbc:p6spy:postgresql://localhost:5432/test

```
    <dependency>
        <groupId>p6spy</groupId>
        <artifactId>p6spy</artifactId>
        <version>2.1.4</version>
    </dependency>
```

**Comment by quintonm** [comment]
try setting excludecategoires as shown below.

`excludecategories=info,debug`

**Comment by Playmate** [comment]
Hm. You know, I've tried, but it's not what I want. Why does not show it like in a batch array or section ?
I think it will be clear to got that all those queries will be executed in one batch or smth like this :
select c from client c where c.id in (1,2);
batch [update
            CLIENT 
        set
            age=20,
            name='Igor',
            nick=NULL 
        where
            id_client=2, update
            CLIENT 
        set
            age=20,
            name='Igor',
            nick=NULL 
        where
            id_client=1 ,
update
            CLIENT 
        set
            age=20,
            name='Igor',
            nick=NULL 
        where
            id_client=2  ]

As I know all batches are stored in ArrayList :
 public void addBatch() throws SQLException
    {
        checkClosed();

        if (batchStatements == null)
        {
            batchStatements = new ArrayList();
            batchParameters = new ArrayList();
        }

        // we need to create copies of our parameters, otherwise the values can be changed
        batchStatements.add(preparedQuery);
        batchParameters.add(preparedParameters.copy());
    }

**Comment by quintonm** [comment]
Did changing excludecategories result in both updates being logged?  

As for the suggested format, an array representation in the log will not work for p6spy.  This is because it logs the JDBC calls as they are made.  Each time addBatch() is called, a log message is generated (assuming that excludecategories does not prevent it).

**Comment by Playmate** [comment]
Let's see how hibernate makes logs in my case :
List<Client> list = em.createQuery("select c from Client c inner join c.accounts", Client.class)
                .getResultList();
        for (Client client : list) {
            client.setName("Igor");
        }
Hibernate: /* select c from Client c inner join c.accounts */ select client0_.id_client as id_clien1_2_, client0_.age as age2_2_, client0_.name as name3_2_, client0_.nick as nick4_2_ from CLIENT client0_ inner join ACCOUNT accounts1_ on client0_.id_client=accounts1_.client_id
Hibernate: /* update com.luxoft.igor.entity.Client */ update CLIENT set age=?, name=?, nick=? where id_client=?
Hibernate: /* update com.luxoft.igor.entity.Client */ update CLIENT set age=?, name=?, nick=? where id_client=?

As you see he generate two update statements, but I know they have been invoked in one batch, but spy generates : two batches and one statement logs, it's no totally clear who does not know about batching.
Actually I've developed my own Formatter :
public class MyFormatter implements MessageFormattingStrategy {

    private static final Formatter BASIC_FORMATTER = new BasicFormatterImpl();

    @Override
    public String formatMessage(int connectionId, String now, long elapsed, String category, String prepared, String sql) {
        if (sql.isEmpty()) {
            return "";
        }
        return "Hibernate: " + BASIC_FORMATTER.format(sql) + " {elapsed: " + elapsed + "ms}";
    }
}

As you see I used BasicFormatterImpl, it's the Hibernate formatter. And it looks the same as before without using spy, but with parameters binding. This log is familiar for developers, but make them happier to view sql parameters and elapsed time !
